src/main/java/dictionary/Dictionary.java

```diff
@@ -0,0 +1,212 @@
+
+            int length = 0;
+            char[] wordArray = word.toCharArray();
+            length = minlength(word, incompleteWord);
+            for (int i = 0; i < length; i++) {
+                if (word.charAt(i) != incompleteWord.charAt(i)) {
+                    if (flag) {
+                        break;
+                    }
+                    listOfCorrectWords.add(completeWord);
+                    flag = true;
+                    break;
+                } else if (i == length - 1) {
+                    listOfCorrectWords.add(word);
+                }
+            }
+        }
+
+        return listOfCorrectWords;
+
+    }
+
+    /**
+     * This  method corrects the word given by user by searching in the dict hashmap
+     *
+     * @param incorrectWord takes input from user
+     * @return the correct word
+     */
+    public static String autocorrect(String incorrectWord) {
+        int diff = Integer.MAX_VALUE;
+        String correct = "Not a single word matches with given words";
+        for (String word : dict.keySet()) {
+
+            int length = 0;
+
+            length = minlength(word, incorrectWord);
+
+            for (int i = 0; i < length; i++) {
+
+                if (word.charAt(i) != incorrectWord.charAt(i)) {
+                    if (i > 0) {
+                        if (Math.abs(incorrectWord.length() - word.length()) < diff) {
+                            correct = word;
+                            diff = Math.abs(incorrectWord.length() - word.length());
+                            break;
+                        }
+                    }
+
+                }
+
+            }
+        }
+        return correct;
+
+
+    }
+
+    public static void main(String args[]) {
+
+        Scanner sc = new Scanner(System.in);
+
+        boolean flag = true;
+        while (flag) {
+            logger.info("Enter 1 to insert in Dictionary");
+            logger.info("Enter 2 to search in Dictionary");
+            logger.info("Enter 3 to autocomplete ");
+            logger.info("Enter 4 to autocorrect ");
+            logger.info("Enter 5 to exit the dictionary");
+            int choice = sc.nextInt();
+
+            switch (choice) {
+                case 1:
+                    System.out.println("enter the word to insert in dictionary");
+                    String word = sc.next();
+                    sc.nextLine();
+                    System.out.println("Enter the meaning of word");
+                    String meaning = sc.nextLine();
+                    insert(word, meaning);
+                    logger.info("Word and meaning is inserted in dictionary\n.......");
+                    break;
+                case 2:
+                    System.out.println("Enter the word for which we want meaning");
+                    String strword = sc.next();
+                    String s = search(strword);
+                    System.out.println("Meaning of given word is " + s);
```

Comment:
what will happen if you are not able to find a word in your dictionary (negative scenarios)?
add a different log level for negative scenarios everywhere.